crates/teloxide/src/dispatching/dispatcher.rs

```diff
@@ -295,15 +295,39 @@ where
     /// [`shutdown`]: ShutdownToken::shutdown
     pub async fn dispatch_with_listener<'a, UListener, Eh>(
         &'a mut self,
-        mut update_listener: UListener,
+        update_listener: UListener,
         update_listener_error_handler: Arc<Eh>,
     ) where
         UListener: UpdateListener + 'a,
         Eh: ErrorHandler<UListener::Err> + 'a,
         UListener::Err: Debug,
+    {
+        self.try_dispatch_with_listener(update_listener, update_listener_error_handler)
+            .await
+            .expect("Couldn't prepare dispatching context")
+    }
+
+    /// Same as `dispatch_with_listener` but returns a `Err(_)` instead of
+    /// panicking when the initial `get_me` call fails.
+    ///
+    /// Starts your bot with custom `update_listener` and
+    /// `update_listener_error_handler`.
+    ///
+    /// This method adds the same dependencies as [`Dispatcher::dispatch`].
+    ///
+    /// [`shutdown`]: ShutdownToken::shutdown
```

Comment:
What is this line for?

Reply:
Oh, that looks like an artifact of the past, all `*dispatch*` functions have this unused link def

Reply:
Should I remove this for all *dispatch* functions as part of this PR ?

Reply:
Yes, sure.

Reply:
done